src/SerializedGameData.cpp

```diff
@@ -247,15 +247,15 @@ void SerializedGameData::MakeSnapshot(const GameWorld& gw)
     writtenEventIds.clear();
 }
 
-void SerializedGameData::ReadSnapshot(GameWorld& gw)
+void SerializedGameData::ReadSnapshot(boost::weak_ptr<Game> game, GameWorld& gw)
```

Comment:
If you pass the game here, you don't need the gw anymore as it is part of game. Also use a shared_ptr (see other comment on LoadMap)

Reply:
Yeah, before i had problems with compiling, thats the reason i keept GameWorld as parameter.
I also moved MakeSnapshop to Game (to keep this API equal), i hope its okay!

Reply:
Just tell me what problems you have and we can resolve them together. I'm happy to help :)